@xen-orchestra/xapi/network.mjs

```
@@ -6,9 +8,35 @@ export default class Network {
     const pluggedVIFs = network.$VIFs.filter(vif => vif.currently_attached && vif.$VM.power_state === 'Running')
     const pluggedPIFs = network.$PIFs.filter(pif => pif.currently_attached)
 
-    await Promise.allSettled(pluggedVIFs.map(vif => this.callAsync('VIF.unplug', vif.$ref)))
-    await Promise.allSettled(pluggedPIFs.map(pif => this.callAsync('PIF.plug', pif.$ref)))
-    await Promise.allSettled(pluggedPIFs.map(pif => this.callAsync('PIF.plug', pif.$ref)))
-    await Promise.allSettled(pluggedVIFs.map(vif => this.callAsync('VIF.plug', vif.$ref)))
+    let errorOccurred = false
+
+    for (const { method, networkInterfaces } of [
+      {
+        method: 'VIF.unplug',
+        networkInterfaces: pluggedVIFs,
+      }, // unplugging PIFs before re-plugging seems unnecessary
+      {
+        method: 'PIF.plug',
+        networkInterfaces: pluggedPIFs,
+      },
+      {
+        method: 'VIF.plug',
+        networkInterfaces: pluggedVIFs,
+      },
+    ]) {
+      try {
+        await asyncEach(networkInterfaces, networkInterface => this.callAsync(method, networkInterface.$ref), {
+          concurrency: 20,
+          stopOnError: false,
+        })
+      } catch (error) {
+        // report an error occured, but continue the re-plugging process
+        errorOccurred = true
```

you should log error here, never eat an error completly
```js
import { createLogger } from '@xen-orchestra/log'
const { warn } = createLogger('xo:xapi:pool')
``` 
And then in the catch:
```js

      warn(error)
      ``